lightning/restore/restore.go

```diff
@@ -1321,6 +1334,80 @@ func (rc *RestoreController) switchTiKVMode(ctx context.Context, mode sstpb.Swit
 	)
 }
 
+func (rc *RestoreController) enforceDiskQuota(ctx context.Context) {
+	if !atomic.CompareAndSwapInt32(&rc.diskQuotaState, diskQuotaStateIdle, diskQuotaStateChecking) {
+		// do not run multiple the disk quota check / import simultaneously.
+		// (we execute the lock check in background to avoid blocking the cron thread)
+		return
+	}
+
+	go func() {
+		rc.diskQuotaLock.Lock()
```

Comment:
Is it better to use an RLock() to check the disk quota? if the quota is exceeded, then change the RLock() to Lock() so we can avoid block all the writers here.

Reply:
🤔 we could move the write-lock next to `logger.Warn("disk quota exceeded")`, though that means the writer are still writing while we are checking disk space (doesn't really matter due to the soft-limit designation). acquiring a read-lock here is useless since it doesn't block anything.